packages/documentation/src/components/Header.jsx

```diff
@@ -1,7 +1,63 @@
+/* eslint-disable react/jsx-key */
 import React from 'react';
-import { Link } from 'gatsby';
+import { Link, graphql } from 'gatsby';
 import sidebarData from '../sidebar';
 import MobileNav from './MobileNav';
+import { Index } from 'elasticlunr';
+// Graphql query used to retrieve the serialized search index.
+export const query = graphql`
+  query SearchIndexExampleQuery {
+    siteSearchIndex {
+      index
+    }
+  }
+`;
+
+// Search component
+class Search extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: ``,
+      results: [],
+    };
+  }
+
+  render() {
+    return (
+      <div>
+        <input type="text" value={this.state.query} onChange={this.search} />
+        <ul>
+          {this.state.results.map(page => (
+            <li>
+              {page.title}: {page.keywords.join(`,`)}
+            </li>
+          ))}
+        </ul>
+      </div>
+    );
+  }
+
+  getOrCreateIndex = () => {
+    return this.index
+      ? this.index
+      : // Create an elastic lunr index and hydrate with graphql query results
+        Index.load(this.props.data.siteSearchIndex.index);
```

Comment:
This blows up.
![Screen Shot 2019-04-24 at 13 09 31](https://user-images.githubusercontent.com/4998130/56686956-6ce95480-6692-11e9-8475-69f2b1bd0914.png)

The documentation doesn't say where this prop is supposed to come from. I traced it through the [example](https://github.com/andrew-codes/gatsby-plugin-elasticlunr-search-demo/blob/6f662e66ce0ad48f1a0bc4bf6579a618958deb29/src/templates/default.js#L8) and it's not clear where any of this data comes from.